hyperledger/sawtooth-core
# Submit CreateRecord transactions from the FishNet client

This PR updates theFishNet client services to allow for submission of CreateRecord transactions (and other more complex T&T transactions). The Add Fish Form has been updated to take advantage of these services, and now actually adds fish Records to the blockchain.

Adding the `'fish'` is done with a script and a batch that is manually submitted through the REST API, but should later be added to the genesis block when there is an official T&T Docker compose file.

**To Test:**
Follow the setup instructions from [PR-879](https://github.com/hyperledger/sawtooth-core/pull/879).

Additionally, start up a REST API on `http://localhost:8008`.

Then, run the batch creation script:
```
cd families/track_and_trade/server/
npm run make-fish
```

And finally submit the batch to the REST API:
```
curl --request POST \
    --header "Content-Type: application/octet-stream" \
    --data-binary @fish.batch \
    "http://localhost:8008/batches"
```

Once the fish type is on the blockchain, you should be able to log into the web app and us it to create new fish to your heart's content.
